### PR TITLE
[MLIR][Linalg] Remove failing obsolete OpDSL test for linalg.matmul.

### PR DESCRIPTION
**Pull request by shahidact**
This PR is related to earlier PR https://github.com/llvm/llvm-project/pull/104783 .

The earlier PR commit was reverted due to a failing OpDSL test for linalg.matmul. Since linalg.matmul is now defined using TableGen ODS instead of Python-based OpDSL, this test is failing and needs to be removed.

This commit removes the failing obsolete test.

### CODE REVIEW

File: mlir/lib/Dialect/Linalg/IR/LinalgOps.cpp [hunk 1]
```diff
@@ -149,6 +149,26 @@ static void fillStructuredOpRegion(OpBuilder &opBuilder, Region &region,
   // iterator_types is an auto-generated method.
 }
 
+/// Helper to create a typical indexing map for MatmulOp. Returns a list of
+/// AffineMap.
+static SmallVector<AffineMap>
+getDefaultIndexingMapsForMatmul(MLIRContext *context) {
```
Comment (MaheshRavishankar) [inline mlir/lib/Dialect/Linalg/IR/LinalgOps.cpp:155]: Why was this added here?

Comment (rengolin) [reply mlir/lib/Dialect/Linalg/IR/LinalgOps.cpp:155]: I think the branch is broken, I don't see the original patch that got reverted. @shahidact can you crate a new branch and cherry-pick the two commits instead of rebasing? git often gets confused when rebasing.

Comment (shahidact) [reply mlir/lib/Dialect/Linalg/IR/LinalgOps.cpp:155]: Closing this PR as another PR(https://github.com/llvm/llvm-project/pull/115319) is raised to this effect.
